auth/credentials/idtoken/idtoken.go

```diff
@@ -113,13 +113,20 @@ func NewCredentials(opts *Options) (*auth.Credentials, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
-	if b := opts.jsonBytes(); b != nil {
-		return credsFromBytes(b, opts)
-	}
-	if metadata.OnGCE() {
+	b := opts.jsonBytes()
+	if b == nil && metadata.OnGCE() {
 		return computeCredentials(opts)
+	} else if b == nil {
+		creds, err := credentials.DetectDefault(&credentials.DetectOptions{
+			Audience: opts.Audience,
```

Comment:
instead of calling the jsonBytes method should we just call DetectDefault and pass in the opts here.

Reply:
`jsonBytes` is called in order to determine if we should skip `DetectDefault` and go immediately into the `idtoken.computeCredentials` path. Thus the precedence is as follows. Let me know if it's wrong.

1. CredentialsFile or CredentialsJSON is present
2. OnGCE == true
3. DetectDefault.

Reply:
Question: If we just call DetectDefault and pass in the opts here, what do we do with the resulting `creds`? If we simply call `credsFromBytes(creds.JSON(), opts)` with it, we are mostly throwing away the opts passed here.

Reply:
I refactored to use `DetectDefault` in all non-compute cases, which simplifies the code a bit.